spec/system/projects_spec.rb

```diff
@@ -1,26 +1,41 @@
 require 'rails_helper'
 
 RSpec.describe "Projects", type: :system do
-  scenario "user creates a new project" do
-    user = FactoryBot.create(:user)
-    # using our custom login helper:
-    # sign_in_as user
-    # or the one provided by Devise:
-    sign_in user
+  let(:user) { FactoryBot.create(:user) }
 
+  before do
+    sign_in user
     visit root_path
+  end
 
-    expect {
-      click_link "New Project"
-      fill_in "Name", with: "Test Project"
-      fill_in "Description", with: "Trying out Capybara"
-      click_button "Create Project"
+  def create_project
+    click_link "New Project"
+    fill_in "Name", with: "Test Project"
+    fill_in "Description", with: "Trying out Capybara"
+    click_button "Create Project"
+  end
 
+  def edit_project
+    click_link "Edit"
+    fill_in "Name", with: "Edit Project"
+    click_button "Update Project"
+  end
+
+  scenario "user creates a new project" do
+    expect {
+      create_project
```

Comment:
`create_project`がどんなデータを作るのか、メソッドの定義を見に行かないとわからないのは非効率です。
（たとえば29行目でなぜ"Test Project"が出てくるのかは、`create_project`を見に行かないとわからない）
なので以下のように引数で入力データを渡す方がベターです。

```suggestion
      create_project(name: 'Test Project', description: 'Trying out Capybara')
```

あとプロジェクト名は"Test Project"や"Edit Project"のような「いかにもテスト用です」というものよりも、実際にありそうなデータの方がテストコードが読みやすくなります。（日本語でもOKですよ）

Reply:
とても腑に落ちます😌